core/bentley/src/TypedArrayBuilder.ts

```diff
@@ -0,0 +1,173 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Collections
+ */
+
+import { Constructor } from "./UtilityTypes";
+import { assert } from "./Assert";
+
+/** Options used to construct a [[TypedArrayBuilder]].
+ * @public
+ */
+export interface TypedArrayBuilderOptions {
+  /** Controls how much additional memory is allocated when the TypedArray needs to be resized to accomodate more elements.
+   * [[TypedArrayBuilder.ensureCapacity]] multiplies the required capacity by this factor to determine the new capacity.
+   * Default: 1.5
+   * Minimum: 1.0, which causes the TypedArray to allocate exactly the space it needs each time it is resized.
+   */
+  growthFactor?: number;
+  /** The number of elements to allocate memory for in the TypedArray when creating the builder.
+   * If you know the minimum number of elements you intend to add to the builder, you should specify that as the
+   * initial capacity to avoid reallocations when populating the array.
+   * Default: 0.
+   * Minimum: 0.
+   */
+  initialCapacity?: number;
+}
+
+/** Incrementally builds a [TypedArray] of unsigned 8-, 16-, or 32-bit integers.
+ * Sometimes you wish to populate a `TypedArray`, but you don't know how many elements you will need.
+ * `TypedArray` requires you to specify the size upon construction, and does not permit you to change the size later.
+ *
+ * `TypedArrayBuilder` manages a `TypedArray`, permitting you to [[push]] and [[append]] elements to it. It exposes two "size" properties":
+ *  - [[capacity]], the number of elements it has currently allocated space for - i.e., the length of the underlying TypedArray; and
+ *  - [[length]], the number of elements that have so far been added to it, which is never bigger than [[capacity]].
+ * When [[capacity]] is exceeded, a new, bigger TypedArray is allocated and the contents of the previous array are copied over to it.
+ *
+ * Once you've finished adding elements, you can obtain the finished `TypedArray` via [[toTypedArray]].
+ * @see [[Uint8ArrayBuilder]], [[Uint16ArrayBuilder]], and [[Uint32ArrayBuilder]] to build specific types of arrays.
+ * @public
+ */
+export class TypedArrayBuilder<T extends Uint8Array | Uint16Array | Uint32Array> {
```

Comment:
Nice class. No tests?

Might be a nice enhancement to allow a "growthFunction" to have more complex growth policies. A future PR.

Reply:
It is indirectly tested by the VertexTableSplitter tests. Are you asking me to add some tests?

Reply:
No, i realize it's tested by your actual use of it in your tests and you want to complete this PR. I just think it's likely that we'll start using this often now that it exists, and tests often serve as examples.